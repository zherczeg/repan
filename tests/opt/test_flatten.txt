# Test flatten optimization

/a(?:(?:(?:)))b/
    flatten, print

/a(?:(?:(?:b|c)))d/
    flatten, print

/a(?:(?:(?:b)|(?:c)|d(?:)))e/
    flatten, print

/(?:ab)(?:)(?:(?:))(?:cd)/
    flatten, print

/a(?:(?s:(?:).))b/
    flatten, print

/(?:ab(?i)(?:c(?s:.))de)|(?i:f)/
    flatten, print

/(ab)(?n:(cd)(ef))(gh)/
    flatten, print
