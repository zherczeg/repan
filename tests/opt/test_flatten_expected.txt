# Test flatten optimization

/a(?:(?:(?:)))b/
    flatten, print
    /ab/

/a(?:(?:(?:b|c)))d/
    flatten, print
    /a(?:b|c)d/

/a(?:(?:(?:b)|(?:c)|d(?:)))e/
    flatten, print
    /a(?:b|c|d)e/

/(?:ab)(?:)(?:(?:))(?:cd)/
    flatten, print
    /abcd/

/a(?:(?s:(?^:.)(?ims:).))b./
    flatten, print
    /a.(?s).b(?-s)./

/(?:ab(?i)(?:c(?s:.))de)|(?i:f)(?-ims:g)/
    flatten, print
    /ab(?i)c(?s).de|f(?-i)g/

/(ab)(?n:(cd)(ef))(gh)/
    flatten, print
    /(ab)cdef(gh)/

# Remove empty alternatives

/(?:|||a||||b|c||d)/
    flatten, print
    /(?:|a||b|c||d)/

/(||||||)/
    flatten, print
    /()/
